src/ExtraConfigurator.php

```diff
@@ -69,37 +65,35 @@ public function setEnvDebugMode()
 	}
 
 	/**
-	 * FACTORIES ***************************************************************
-	 */
-
-	/**
-	 * Setup debug mode and add parameters immediately
-	 *
-	 * @param Configurator $configurator
-	 * @return Configurator
+	 * @param string $fileName
+	 * @return void
 	 */
-	public static function setup(Configurator $configurator)
+	public function setFileDebugMode($fileName = NULL)
 	{
```

Comment:
It's nice. What about little tunning?

```
function setFileDebugMode($fileName = null)
{
	$appDir = $this->parameters['appDir'] ? $this->parameters['appDir'] : null;

	// No filename and no application root dir
	if (!$fileName && !$appDir) return;

	$content = @file_get_contents($fileName ?: $appDir . '/../.debug');
	if ($content === false || empty($content)) {
		$this->setDebugMode(true);
	} else {
		$debug = self::parseDebugValue(trim($content));
		$this->setDebugMode($debug);
	}

}
```

Reply:
@f3l1x seems file besides. `empty($content)` cant be used cause its invalid for "0" etc

Reply:
I see, `$content === ''` would be fine.